bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/icon/FontAwesome5IconType.java

```diff
@@ -0,0 +1,1268 @@
+package de.agilecoders.wicket.extensions.markup.html.bootstrap.icon;
+
+import de.agilecoders.wicket.core.markup.html.bootstrap.image.IconType;
+import de.agilecoders.wicket.extensions.markup.html.bootstrap.icon.FontAwesome5IconTypeBuilder.FontAwesome5Brand;
+import de.agilecoders.wicket.extensions.markup.html.bootstrap.icon.FontAwesome5IconTypeBuilder.FontAwesome5Regular;
+import de.agilecoders.wicket.extensions.markup.html.bootstrap.icon.FontAwesome5IconTypeBuilder.FontAwesome5Solid;
+import org.apache.wicket.util.string.Strings;
+
+import static de.agilecoders.wicket.extensions.markup.html.bootstrap.icon.FontAwesome5IconTypeBuilder.on;
+
+/**
+ * All the icon types.
+ */
+public class FontAwesome5IconType extends IconType {
```

Comment:
I have always wondered how people create these classes ?
Did you create it manually ? Or with a script ?
I ask because I guess one day I will have to update it.

Reply:
It wasn't one of my favorite chores either :-) I copied all the icon names from the fontawesome page (https://fontawesome.com/icons?d=gallery&m=free filtering by style) and pasted that into a reasonable editor.

I then used regex replacement to get from 

```
ad
address_book
```

to

```
    public static final FontAwesome5IconType ad_s = on(FontAwesome5Solid.ad).build();
    public static final FontAwesome5IconType address_book_r = on(FontAwesome5Regular.address_book).build();
```
and just pasted it into the class file I had copied from the v4 version in the project.

Reply:
Regex FTW! :-)